acceptance/tests/modules/install/long_path_name.rb

```diff
@@ -0,0 +1,34 @@
+test_name "puppet module install (long file path)"
+require 'puppet/acceptance/module_utils'
+extend Puppet::Acceptance::ModuleUtils
+
+confine :except, :platform => /centos-4|el-4/ # PUP-5226
+confine :except, :platform => /aix/ # PUP-5501
+
+module_author = 'puppetlabs'
+module_name   = 'dsc'
+module_version = '1.0.0'
+
+orig_installed_modules = get_installed_modules_for_hosts hosts
+teardown do
+  rm_installed_modules_from_hosts orig_installed_modules, (get_installed_modules_for_hosts hosts)
+end
+
+agents.each do |agent|
+  step 'setup'
+  stub_forge_on(agent)
+
+  step "install module '#{module_author}-#{module_name}'"
+  on(agent, puppet("module install #{module_author}-#{module_name} --version #{module_version}")) do |result|
+    @module_path = /(Notice: Preparing to install into )(.*)( \.\.\.)/.match(result.stdout)[2]
+    assert_module_installed_ui(stdout, module_author, module_name)
+  end
+
+  path = "#{@module_path}/#{module_name}"
+  path = on(agent, "cygpath -u #{path}").stdout.chomp if /win/ =~ agent.platform
+
+  step 'Count the files extracted from the tar'
+  on(agent, "find #{path} -not -path '*/\\.*' -type f | wc -l") do |result|
+    assert_match(/2602/, result.stdout, 'The correct number of files was not observed')
```

Comment:
I'm not disagreeing about the magic number removal - I was talking about what would need to be done to facilitate that.

* You probably want to have access to the tarball to get the file list / count on the coordinator side
* Windows agents probably shouldn't introspect the tarball:
  * `tar` command in in Cygwin, but not elsewhere... I think you would have to pipe the extraction through `| wc`, which is a little weird
  * If using a Ruby script on Windows that would require using minitar to examine contents (not recommended given that's where the extract bug presented)
  * It's a bit weird to ask agent for count of files in tarball, to then plug that back into the coordinator side, to then ask the agent to verify

Therefore, 2 options:

* Pull the `.tar.gz` from fake forge / Internet to coordinator, and examine
* Have the `.tar.gz` in this repo, which reduces network dependency (the comment about file size was to not bloat this git repo with large binaries)

IMHO, there's no good reason to make an extra pull over the network to grab the module, as copying it via SCP from Beaker coordinator to agents and then calling `puppet module install tarball.tar.gz` achieves the same result.